frodo.js

```diff
@@ -1,18 +1,20 @@
 /**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.